src/ArrangerConduit.sol

```diff
@@ -0,0 +1,256 @@
+            asset:           asset,
+            ilk:             ilk,
+            amountRequested: amount,
+            amountFilled:    0,
+            info:            info
+        }));
+
+        requestedFunds[ilk][asset] += amount;
+        totalRequestedFunds[asset] += amount;
+
+        emit RequestFunds(ilk, asset, fundRequestId, amount, info);
+    }
+
+    function cancelFundRequest(uint256 fundRequestId) external override {
+        FundRequest memory fundRequest = fundRequests[fundRequestId];
+
+        address asset = fundRequest.asset;
+        bytes32 ilk   = fundRequest.ilk;
+
+        _checkAuth(ilk);
+
+        uint256 amountRequested = fundRequest.amountRequested;
+
+        delete fundRequests[fundRequestId];
+
+        requestedFunds[ilk][asset] -= amountRequested;
+        totalRequestedFunds[asset] -= amountRequested;
+
+        emit CancelFundRequest(fundRequestId);
+    }
+
+    /**********************************************************************************************/
+    /*** Fund Manager Functions                                                                 ***/
+    /**********************************************************************************************/
+
+    function drawFunds(address asset, uint256 amount) external override isArranger {
+        require(amount <= drawableFunds(asset), "Conduit/insufficient-funds");
+
+        require(ERC20Like(asset).transfer(arranger, amount), "Conduit/transfer-failed");
+
+        emit DrawFunds(asset, amount);
+    }
+
+    function returnFunds(uint256 fundRequestId, uint256 returnAmount)
+        external override isArranger
+    {
+        FundRequest storage fundRequest = fundRequests[fundRequestId];
+
+        require(fundRequest.status == StatusEnum.PENDING, "Conduit/invalid-status");
+
+        address asset = fundRequest.asset;
+        bytes32 ilk   = fundRequest.ilk;
+
+        uint256 amountRequested = fundRequest.amountRequested;
+
+        withdrawableFunds[ilk][asset] += returnAmount;
+        totalWithdrawableFunds[asset] += returnAmount;
+
+        requestedFunds[ilk][asset] -= amountRequested;
+        totalRequestedFunds[asset] -= amountRequested;
+
+        fundRequest.amountFilled += returnAmount;
+
+        fundRequest.status = StatusEnum.COMPLETED;
+
+        require(
+            ERC20Like(fundRequest.asset).transferFrom(arranger, address(this), returnAmount),
+            "Conduit/transfer-failed"
+        );
+
+        emit ReturnFunds(ilk, asset, fundRequestId, amountRequested, returnAmount);
+    }
+
+    /**********************************************************************************************/
+    /*** View Functions                                                                         ***/
+    /**********************************************************************************************/
+
+    function drawableFunds(address asset) public view returns (uint256 drawableFunds_) {
```

Comment:
This should be included in the `IArrangerConduit` interface.

Reply:
Updated